doc/source/dev/howto-docs.rst

```diff
@@ -215,6 +215,218 @@ Note that for documentation within NumPy, it is not necessary to do
 Please use the ``numpydoc`` :ref:`formatting standard <numpydoc:format>` as
 shown in their :ref:`example <numpydoc:example>`.
 
+.. _doc_c_code:
+
+Documenting C/C++ Code
+======================
+
+NumPy uses Doxygen_ to parse specially-formatted C/C++ comment blocks. This generates
+XML files, which are  converted by Breathe_ into RST, which is used by Sphinx.
+
+**It takes three steps to complete the documentation process**:
+
+1. Writing the comment blocks
+-----------------------------
+
+Although there is still no commenting style set to follow, the Javadoc
+is more preferable than the others due to the similarities with the current
+existing non-indexed comment blocks.
+
+.. note::
+   Please see `"Documenting the code" <https://www.doxygen.nl/manual/docblocks.html>`__.
+
+This is how Javadoc style looks like::
+
+  /**
+   * Your brief is here.
+   */
+  void function(void);
+
+For line comment, insert a triple forward slash::
+
+  /// Your brief is here.
+  void function(void);
+
+And for structured data members, just insert triple forward slash and less-than sign::
+
+  /// Data type brief description
+  typedef struct {
+      int member_1; ///< Member 1 description.
```

Comment:
Similar argument against `///<`, it is not immediately clear where in the output this goes.

Reply:
data members shouldn't have brief and details from the first place. just general description since they're not supposed to be indexed.

Reply:
In which case this section should be removed.

Reply:
Improve it with a better example, please check the new updates.